pkg/ebpftracer/c/tracee.bpf.c

```diff
@@ -6267,6 +6271,94 @@ statfunc int net_l7_is_http(struct __sk_buff *skb, u32 l7_off)
     return 0;
 }
 
+// clang-format on
+
+#define SOCKS5_VERSION(buf)     buf[0]
+#define SOCKS5_NUM_METHODS(buf) buf[1]
+#define SOCKS5_CMD(buf)         buf[1]
+#define SOCKS5_RESERVED(buf)    buf[2]
+#define SOCKS5_ADDR_TYPE(buf)   buf[3]
+
+// see https://datatracker.ietf.org/doc/html/rfc1928 for the definition of the socks5 protocol
+statfunc bool net_l7_is_socks5(struct __sk_buff *skb, u32 l7_off)
```

Comment:
We only need socks5 connect message. From what I see now it will output all packets.

Reply:
Sadly the socks5 connect message is not enough, as from it alone we cannot distinguish it from the servers response. Overall this should not be too big of a deal though, as the number of socks5 messages should be rather small (only required when opening the connection).

Reply:
Ok, make sense.